Src/CSharpier/SyntaxPrinter/ConstraintClauses.cs

```diff
@@ -16,19 +16,40 @@ public static Doc Print(IEnumerable<TypeParameterConstraintClauseSyntax> constra
             {
                 return Doc.Null;
             }
+            else if (constraintClausesList.Count == 1)
```

Comment:
So, this whole function can be replaced by C# 8/9's new pattern matching syntax :rocket: 
```
return constraintClauses.ToList() switch
{
    { Count: 0 } => Doc.Null,
    { Count: 1 } => /* stuff */,
    var list when list[0].Parent is MethodDeclarationSyntax => /* stuff */,
    var list => Doc.Indent(
        Doc.HardLine,
        Doc.Join(Doc.HardLine, list.Select(TypeParameterConstraintClause.Print))
    )                                               
};
```